crates/bevy_render/src/lib.rs

```diff
@@ -286,14 +293,14 @@ impl Plugin for RenderPlugin {
                     render_app.world.clear_entities();
                 }
             });
-        }
+        });
 
+        // Only add these plugins after adding the above render_init function, to ensure
+        // any render initialization required for these plugins will find the sub RenderApp
         app.add_plugin(WindowRenderPlugin)
             .add_plugin(CameraPlugin)
             .add_plugin(ViewPlugin)
             .add_plugin(MeshPlugin)
-            // NOTE: Load this after renderer initialization so that it knows about the supported
-            // compressed texture formats
```

Comment:
I'd prefer to keep this comment as I feel the dependency is less obvious and it clarifies why for future readers.